src/libraries/System.IO.Compression/tests/ZipArchive/zip_CreateTests.cs

```diff
@@ -196,5 +197,44 @@ private static void AssertDataDescriptor(MemoryStream memoryStream, bool hasData
             Assert.Equal(hasDataDescriptor ? 8 : 0, fileBytes[6]);
             Assert.Equal(0, fileBytes[7]);
         }
+
+        [Theory]
+        // General purpose bit flag must get the appropriate bit set if a file comment or an entry name is unicode
+        [InlineData("ascii", "ascii!!!", "utf-8", "U��������")]
+        [InlineData("utf-8", "U��������", "ascii", "ascii!!!")]
+        [InlineData("ascii", "ascii!!!", "latin1", "L��������")]
+        [InlineData("latin1", "L��������", "ascii", "ascii!!!")]
+        [InlineData("utf-8", "U��������", "latin1", "L��������")]
+        [InlineData("latin1", "L��������", "utf-8", "U��������")]
+        public static void Create_ZipArchiveEntry_DifferentEncodingsFullNameAndComment(string en1, string s1, string en2, string s2)
+        {
+            Encoding e1 = Encoding.GetEncoding(en1);
+            Encoding e2 = Encoding.GetEncoding(en2);
+            string entryName = e1.GetString(e1.GetBytes(s1));
+            string comment = e2.GetString(e2.GetBytes(s2));
+
+            var stream = new MemoryStream();
+            var testStream = new WrappedStream(stream, true, true, true, null);
+
+            // Create with no encoding to autoselect it if one of the two strings is unicode
+            using (var zip = new ZipArchive(testStream, ZipArchiveMode.Create, leaveOpen: true))
+            {
+                ZipArchiveEntry entry = zip.CreateEntry(entryName, CompressionLevel.NoCompression);
+                entry.Comment = comment;
+
+                Assert.Equal(entryName, entry.FullName);
+                Assert.Equal(comment, entry.Comment);
+            }
+
+            // Open with no encoding
+            using (var zip = new ZipArchive(testStream, ZipArchiveMode.Read, leaveOpen: false))
```

Comment:
I am surprised that we don't need to Seek to `testStream` beginning.

Reply:
https://github.com/dotnet/runtime/blob/7a62468ffbb90008e08b30471bfa94b2d4c58726/src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchive.cs#L138-L147